### PR TITLE
Improve RedisSessionStorage.findSessionsByShop:  Do Not Fail if Non-Session Data Exists in Redis 

### PR DESCRIPTION
**Pull request by devstein**
### WHY are these changes introduced?

Right now, `findSessionsByShop` throws an exception if the Redis instance has non-session data that isn't parse-able by `JSON.parse` or `sessionFromEntries`

### WHAT is this pull request doing?

This PR skips values (`rawResult`) stored in Redis that are not related to shop sessions.

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above
- [ ] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

**Comment by devstein** [comment]
Hi 👋 , any recommendations for testing? This edge-case is implementation specific and doesn't fit into the `SessionStorage` interface (i.e as a new test in battery-of-tests)

**Comment by devstein** [comment]
Signed the CLA

File: src/auth/session/storage/redis.ts [hunk 1]
```diff
@@ -85,13 +85,19 @@ export class RedisSessionStorage implements SessionStorage {
...
       if (!rawResult) continue;
 
+      try {
+
       const session = sessionFromEntries(JSON.parse(rawResult));
-      if (session.shop === cleanShop) results.push(session);
+      if (session?.shop === cleanShop) results.push(session);
+      } 
+      // do nothing if the rawResult is not a parse-able session
+      catch {}
```
Comment (cquemin) [inline src/auth/session/storage/redis.ts:100]: Should we catch the Json Error raised here or any more relevant parsing exception? I am conscious of hiding other exceptions that we may want to let through if they occur. I know this would require some custom code in order to get the type of the exception or re throw it if it is not what we want.
Not sure if this is very JS idiomatic but `catch all` blocks always scare me :)

Comment (devstein) [reply src/auth/session/storage/redis.ts]: Narrowing error types isn't elegant in JS, but gave it go. I agree that the `catch all` is scary

**Review by devstein — COMMENTED** [review]
@cquemin Updated. Let me know what you think
